### ISSUE TITLE
Add user forms when Tasks are activated.  Use the field user_form_id to look up the template in the table user_form.

### ISSUE THREAD

**Issue by FWAJL** [FWAJL/FieldWorkAssistantMVC#906]
@bsaiken @gsouvik 

From @bsaiken's comment (https://github.com/FWAJL/FieldWorkAssistantMVC/issues/727#issuecomment-95895018):

Can you complete the details so Souvik knows what is involved.

Thank you.

**Comment by bsaiken** [comment]
@FWAJL 
Why would we ignore user forms?

**Comment by FWAJL** [comment]
@bsaiken @gsouvik 

Because of what you said here (in bold from #727):

> Don't get too generic about copying forms. We will eventually have to deal with the case where the template is only copied once. For example, the chain of custody, which will be a commonly used form, will not get copied once for each location. Instead, all the location names will be put on the form. ***I do NOT want to implement this now, but it is important to know.***

If a task form is a user form then it halts the execution. Therefore we need to add a bit of code to prevent that hence this issue.

**Comment by bsaiken** [comment]
@FWAJL @gsouvik 
I think you have a misunderstanding of the difference between a user for and a master form.  Both should be copied as many times as there are locations.  

Your statement `If a task form is a user form then it halts execution` is incorrect.

We will not make one copy for every location in all instances.  The best example is the chain of custody form.  That form will only have one copy, and the (single) form will list all the locations.  So my comment was meant to code in such a way that we could later add an identifier for forms that get coped only once, like the chain of custody form.  If I have not explained clearly enough, let's talk about it Wednesday, but this Issue is not needed and we will close it.

**Comment by gsouvik** [comment]
@bsaiken @FWAJL 
Just wanted to add my input to this discussion, at this moment we are not explicitly filtering user forms from the table `task_template_form`. So all the forms are being pulled from that relationship table (including the user forms too). But when the corresponding template form is looked up, it is currently being done from only the master forms (and NOT from the `user_form` table), and that produced the issue while @bsaiken tested. So the ideal solution would be either of the following:

1) Considering user forms too and based on the field `user_form_id` look up the template in the table `user_form`.
2) If user form section is not ready yet, change code a bit to NOT consider the records based on the field `user_form_id` as @FWAJL suggested. In that case this issue might have some relevance.

Let us know your thoughts. Thank you.

**Comment by bsaiken** [comment]
@gsouvik @FWAJL 
Option 1 is the correct choice.  So, I have modified the name of this Issue to reflect that.

**Comment by gsouvik** [comment]
@bsaiken 
Considering the fact that #778 and #727 which are both not resolved due to this issue, I'm starting with this next even though it has low priority than other tasks. Please confirm.
Thanks.

**Comment by gsouvik** [comment]
@bsaiken @FWAJL 
When task forms are copied into location specific PDFs, user forms are now taken into account. Steps to test are same as before, just promote a task to active which has both user and master forms assigned to it.
Thank you.

**Comment by bsaiken** [comment]
@FWAJL @gsouvik 
I have to push this directly to code review, because I am unable to get this working on my local machine.  I had the same problem with master forms, and it works just great on the live site.  I am hopeful the problem will correct itself, eventually, as my local site and database is updated.  But I have already spent too much time trying to chase down this problem.

**Comment by WebDevJL** [comment]
@gsouvik 

Please look at the coding update I made in the function that you made to create task location forms. 

I fragmented the function into smaller functions to get a state where each function does one thing. It does exactly the same thing at the end. 

Tell me if you understand this as it helps us reusing functions later on. 

As a challenge, I'd like you to think about it when coding. It'd be very helpful and I know you are skilled to achieve this little challenge :+1: 

Thank you.

**Comment by gsouvik** [comment]
@WebDevJL 
Yes, you refactored the code to greater granular level which is great when we plan to reuse it for other calls as well. I already implement this approach, but shall make future functions even more modular as per this example. 

Thank you.
